### DISCUSSION TITLE
Influence Diagrams Solvers

### DISCUSSION THREAD

**Discussion by DomDF** [gamma-opt/DecisionProgramming.jl#33] [General] [answered]
Hi all, 

I am interested in using the `DecisionProgramming.jl` library for creating influence diagrams.

In the documentation for this (https://gamma-opt.github.io/DecisionProgramming.jl/dev/examples/used-car-buyer/) all of the examples using `Gurobi.jl` library as the solver (which I understand is an interface for a separate software), for example:

```
optimizer = optimizer_with_attributes(
    () -> Gurobi.Optimizer(Gurobi.Env()),
    "IntFeasTol"      => 1e-9,
)
set_optimizer(model, optimizer)
optimize!(model)
```

I was wondering if it was possible to use other solvers for these problems?

Kind Regards,
Dom

**Comment by FabsOliveira** [comment, marked as answer]
Hi @DomDF. Yes, of course. In principle, you can use any mixed-integer programming solver compatible with JuMP.jl. Basically, any supporting MILP from this [list](https://jump.dev/JuMP.jl/stable/installation/#Supported-solvers ) would work.

Personally, I recommend [HiGHS.jl](https://github.com/jump-dev/HiGHS.jl). It is a great open-source solver that has remarkable performance. Then all you need to do is

```julia
using HiGHS
using JuMP

model = Model(HiGH.Optimizer())
z = DecisionVariables(model, diagram)
x_s = PathCompatibilityVariables(model, diagram, z)
EV = expected_value(model, diagram, x_s)
@objective(model, Max, EV)
optimize!(model)
```  

P.S>: Apologies for the delay. I noticed my notifications from Github were off. We should be more responsive now!

**Reply by DomDF** [reply]
Hi @FabsOliveira - thanks for getting back to me.

Being able to use any solver is great news. I had tried this (using `GLPK.jl`) but wasn't able to figure out the syntax as all the examples used `Gurobi.jl`.

Your example was very helpful, but I needed to change the line that defines the model, as shown below (for anyone who is interested):
```
model = Model(HiGHS.Optimizer)
```

Thank you for getting back to me! 
My research will continue to use influence diagrams and I am keen to do these calculations in Julia, so I may bother you again with future questions about this library.
